synapse/lib/types.py

```diff
@@ -481,6 +481,44 @@ def indxByEq(self, valu):
 
         return Type.indxByEq(self, valu)
 
+
+class Range(Type):
+
+    _opt_defs = {
+        ('subtype', None),
+    }
+
+    def postTypeInit(self):
+        self.subtype = self.opts.get('subtype')
+        if not self.subtype:
+            raise s_exc.BadTypeDef()
+
+        self.setNormFunc(tuple, self._normPyTuple)
+
+    def _normPyTuple(self, valu):
+        if len(valu) is not 2:
+            raise s_exc.BadTypeValu(valu, mesg=f'Must be a 2-tuple of type {self.subtype}')
+
+        t = self.modl.type(self.subtype)
+        v0 = t.norm(valu[0])[0]
+        v1 = t.norm(valu[1])[0]
+
+        if v0 > v1:
```

Comment:
we should do "left side is min, right side is max" that way if v0 > v1 we throw an exception

Reply:
Ya, that's a good point. I'll change this